homeassistant/components/system_bridge/sensor.py

```diff
@@ -0,0 +1,346 @@
+ATTR_SERVICE_PACK = "service_pack"
+ATTR_SIZE = "size"
+ATTR_SPEED = "speed"
+ATTR_SPEED_CURRENT_MAX = "speed_current_max"
+ATTR_SPEED_CURRENT_MIN = "speed_current_min"
+ATTR_SPEED_MAX = "speed_max"
+ATTR_SPEED_MIN = "speed_min"
+ATTR_TEMPERATURE_MAX = "temperature_max"
+ATTR_TIME_REMAINING = "time_remaining"
+ATTR_TYPE = "type"
+ATTR_USED = "used"
+ATTR_VENDOR = "vendor"
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up System Bridge sensor based on a config entry."""
+    coordinator: DataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    bridge: Bridge = coordinator.data
+
+    async_add_entities(
+        [
+            BridgeBatterySensor(coordinator, bridge),
+            BridgeCpuSpeedSensor(coordinator, bridge),
+            BridgeCpuTemperatureSensor(coordinator, bridge),
+            *[
+                BridgeFilesystemSensor(coordinator, bridge, key)
+                for key, _ in bridge.filesystem.fsSize.items()
+            ],
+            BridgeOsSensor(coordinator, bridge),
+            BridgeProcessesLoadSensor(coordinator, bridge),
+        ],
+        True,
+    )
+
+
+class BridgeSensor(BridgeDeviceEntity):
+    """Defines a System Bridge sensor."""
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        bridge: Bridge,
+        key: str,
+        name: str,
+        icon: str,
+        device_class: str = "",
+        unit_of_measurement: str = "",
+    ) -> None:
+        """Initialize System Bridge sensor."""
+        self._device_class = device_class
+        self._unit_of_measurement = unit_of_measurement
+
+        super().__init__(coordinator, bridge, key, name, icon)
+
+    @property
+    def device_class(self) -> Optional[str]:
+        """Return the class of this sensor."""
+        return self._device_class
+
+    @property
+    def unit_of_measurement(self) -> str:
+        """Return the unit this state is expressed in."""
+        return self._unit_of_measurement
+
+
+class BridgeBatterySensor(BridgeSensor):
+    """Defines a Battery sensor."""
+
+    def __init__(self, coordinator: DataUpdateCoordinator, bridge: Bridge):
+        """Initialize System Bridge sensor."""
+        super().__init__(
+            coordinator,
+            bridge,
+            "battery",
+            "Battery",
+            None,
+            DEVICE_CLASS_BATTERY,
+            PERCENTAGE,
+        )
+
+    @property
+    def state(self) -> float:
+        """Return the state of the sensor."""
+        bridge: Bridge = self.coordinator.data
+        return bridge.battery.percent
+
+    @property
+    def device_state_attributes(self) -> Optional[Dict[str, Any]]:
+        """Return the state attributes of the entity."""
+        bridge: Bridge = self.coordinator.data
+        return {
+            ATTR_CAPACITY_MAX: bridge.battery.maxCapacity,
+            ATTR_CAPACITY: bridge.battery.currentCapacity,
```

Comment:
What's capacity and capacity max?

Reply:
Not 100% sure, it's what gets reported from the systeminformation node module:

https://systeminformation.io/battery.html

Reply:
https://github.com/sebhildebrandt/systeminformation/blob/master/lib/battery.js#L69